src/encode.rs

```diff
@@ -79,3 +83,134 @@ fn test_encode_zig_zag() {
     assert_eq!(encode_zig_zag(i64::min_value()), 0xFFFFFFFF_FFFFFFFF);
     assert_eq!(encode_zig_zag(i64::max_value()), 0xFFFFFFFF_FFFFFFFE); // 0x7FFF...FF
 }
+
+pub fn encode<W: Write>(writer: &mut W, schema: &Schema, value: &Value) -> Result<(), EncodeError> {
+    match (schema, value) {
+        (&Schema::Null, &Value::Null) => Ok(()),
+        (&Schema::Null, _)            => Err(EncodeError { kind: EncodeErrorKind::InvalidSchema }),
+
+        (&Schema::Boolean, &Value::Boolean(true))  => { writer.write(&[0x01]); Ok(()) },
```

Comment:
Could you wrap `try!()` around the `write()` call here and in the rest of the function?